### PR TITLE
Fix compiling wasm2c's runtime library with Android NDK(macro definition conflicts with sysroot)

### PR DESCRIPTION
**Pull request by squk**
Building wasm2c'c wasm_rt with the Android NDK results in the following error
```
third_party/wabt/wasm2c/wasm-rt-mem-impl.c:28:9: error: 'PAGE_SIZE' macro redefined [-Werror,-Wmacro-redefined]
#define PAGE_SIZE 65536
        ^
third_party/android/ndk/stable/toolchains/llvm/prebuilt/linux-x86_64/bin/../real-bin/../sysroot/usr/include/sys/user.h:37:9: note: previous definition is here
#define PAGE_SIZE 4096
        ^
1 error generated.
```

This PR fixes that.

### CODE REVIEW

File: wasm2c/wasm-rt-mem-impl.c [hunk 1]
```diff
@@ -25,7 +25,7 @@
 #include <sys/mman.h>
 #endif
 
-#define PAGE_SIZE 65536
+#define WASM_RT_PAGE_SIZE 65536
```
Comment (sbc100) [inline wasm2c/wasm-rt-mem-impl.c:28]: How about WASM_PAGE_SIZE.. since otherwise it might be confused with the host (runtime) page size?

**Comment by sbc100** [comment]
@mathiasblume did you run into this?

**Comment by squk** [comment]
@sbc100 IIRC @matthias-blume  doesn't use the wasm runtime in his experiments and has forked the c-writer

File: wasm2c/wasm-rt-mem-impl.c (hunk 1 above)
Comment (squk) [reply wasm2c/wasm-rt-mem-impl.c:28]: Done.

**Review by sbc100 — APPROVED** [review]

**Comment by sbc100** [comment]
Looks like there is a separate cmake issue in CI that we should fix before landing this.

**Comment by sbc100** [comment]
Rebasing on top of the main branch should fix the current CI failure.

**Comment by squk** [comment]
@sbc100 done :)

**Comment by squk** [comment]
Thanks!

**Pull request merged**